test/jdk/java/lang/ClassLoader/loadLibraryUnload/LoadLibraryUnload.java

```diff
@@ -0,0 +1,159 @@
+ * Copyright (c) 2021, BELLSOFT. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * LoadLibraryUnload class calls ClassLoader.loadedLibrary from multiple threads
+ */
+/*
+ * @test
+ * @bug 8266310
+ * @summary deadlock while loading the JNI code
+ * @library /test/lib
+ * @build LoadLibraryUnload p.Class1
+ * @run main/othervm/native -Xcheck:jni LoadLibraryUnload
+ */
+import jdk.test.lib.Asserts;
+import java.lang.*;
+import java.lang.reflect.*;
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import p.Class1;
+
+public class LoadLibraryUnload {
+
+    private static class TestLoader extends URLClassLoader {
+        public TestLoader() throws Exception {
+            super(new URL[] { Path.of(System.getProperty("test.classes")).toUri().toURL() });
+        }
+
+        @Override
+        public Class<?> loadClass(String name) throws ClassNotFoundException {
+            synchronized (getClassLoadingLock(name)) {
+                Class clazz = findLoadedClass(name);
+                if (clazz == null) {
+                    try {
+                        clazz = findClass(name);
+                    } catch (ClassNotFoundException ignore) {
+                    }
+                    if (clazz == null) {
+                        clazz = super.loadClass(name);
+                    }
+                }
+                return clazz;
+            }
+        }
+    }
+
+    private static class LoadLibraryFromClass implements Runnable {
+        Object object;
+        Method method;
+
+        public LoadLibraryFromClass(Class<?> fromClass) {
+            try {
+                this.object = fromClass.newInstance();
+                this.method = fromClass.getDeclaredMethod("loadLibrary");
+            } catch (Exception error) {
+                throw new Error(error);
```

Comment:
Error is fine.  Most tests throw `RuntimeException` that can be another choice.

Reply:
Yes, used RuntimeException.